Microsoft.Toolkit.Parsers/Markdown/Inlines/BoldItalicTextInline.cs

```diff
@@ -49,14 +51,14 @@ internal static void AddTripChars(List<Common.InlineTripCharHelper> tripCharHelp
         /// <param name="start"> The location to start parsing. </param>
         /// <param name="maxEnd"> The location to stop parsing. </param>
         /// <returns> A parsed bold text span, or <c>null</c> if this is not a bold text span. </returns>
-        internal static Common.InlineParseResult Parse(string markdown, int start, int maxEnd)
+        internal static InlineParseResult Parse(string markdown, int start, int maxEnd)
         {
             if (start >= maxEnd - 1)
             {
                 return null;
             }
 
-            if (markdown == null || markdown.Length < 6)
+            if (markdown == null || markdown.Length < 6 || start + 6 >= maxEnd)
```

Comment:
What's the '6' for?

Reply:
It ensures that the length of the string can possibly fit two sets of "\*\*\*", which is required for Bold and Italic parsing. Without these, creating a BoldItalicInline will crash if prepended or appended to the text, due to the substring calls.